File: app/src/main/java/com/wiseassblog/samsaradayplanner/SamsaraApp.java
package com.wiseassblog.samsaradayplanner;

import android.app.Application;

public class SamsaraApp extends Application {

    private StorageServiceLocator locator;

    @Override
    public void onCreate() {
        super.onCreate();

        locator = new StorageServiceLocator(this);
    }

    public StorageServiceLocator getServiceLocator() {
        return locator;
    }
}

File: app/src/main/java/com/wiseassblog/samsaradayplanner/SamsaraApp.kt
package com.wiseassblog.samsaradayplanner

import android.app.Application

class SamsaraApp : Application() {
    lateinit var serviceLocator: StorageServiceLocator

    override fun onCreate() {
        super.onCreate()
        serviceLocator = StorageServiceLocator(this)
    }
}